netplugin/netd.go

```diff
@@ -654,32 +656,39 @@ func handleStateEvents(netPlugin *plugin.NetPlugin, crt *crt.CRT, opts cliOpts,
 func handleEvents(netPlugin *plugin.NetPlugin, crt *crt.CRT, opts cliOpts) error {
 	recvErr := make(chan error, 1)
 	recvEventErr := make(chan error)
-
-	//monitor and process state change events
 	go handleStateEvents(netPlugin, crt, opts, recvErr)
-	startDockerEventPoll(netPlugin, crt, recvEventErr, opts)
 
-	go func() {
+	//monitor and process state change events
+	if opts.nativeInteg {
 		for {
-			err := <-recvEventErr
-			if err != nil {
-				log.Warnf("Failure occured talking to docker events. Sleeping for a second and retrying. Error: %s", err)
+			<-recvErr
```

Comment:
I think we discussed this on slack a few days ago. It's ok to error out if an error is received on `recvErr` channel. This will make error handling consistent with the check at line 685 below.